community/cypher/cypher/src/test/scala/org/neo4j/cypher/EagerizationAcceptanceTest.scala

```diff
@@ -929,9 +907,9 @@ class EagerizationAcceptanceTest extends ExecutionEngineFunSuite with TableDrive
     relate(node0, node3, "KNOWS")
 
     val query = "MATCH (a:Foo), (b:Bar) MERGE (a)-[r:KNOWS]->(b) ON MATCH SET b:Foo"
-
-    assertStats(executeWithRulePlanner(query), relationshipsCreated = 3, labelsAdded = 1)
-    assertNumberOfEagerness(query, 0)
+    assertStats(updateWithBothPlanners(query), relationshipsCreated = 3, labelsAdded = 1)
+    //TODO this we need to consider not only overlap but also what known labels the node we set has
+    assertNumberOfEagerness(query, 1)
```

Comment:
I updated my badly copy pasted query in the comment. What is said is still true, the regression is already there on these types of queries)